src/RichTextContent.tsx

```diff
@@ -1,14 +1,15 @@
-import { Box } from "@mui/material";
+import { Box, type BoxProps } from "@mui/material";
 import { EditorContent } from "@tiptap/react";
 import { useMemo } from "react";
 import type { CSSObject } from "tss-react";
 import { makeStyles } from "tss-react/mui";
+import type { Except } from "type-fest";
 import { useRichTextEditorContext } from "./context";
 import { getEditorStyles, getUtilityClasses } from "./styles";
 
 export type RichTextContentClasses = ReturnType<typeof useStyles>["classes"];
 
-export type RichTextContentProps = {
+export type RichTextContentProps = Except<BoxProps, "children"> & {
```

Comment:
Should exclude `component` in addition to `children`, since it's always set directly below